woocommerce/woocommerce-ios
# Migrate `Settings` screen to MVVM

Closes #4142 

#4142 proposes to migrate settings screen to MVVM pattern to make it testable. 

This pull request attempts to migrate it to MVVM and add basic tests.
![SettingsScreen](https://user-images.githubusercontent.com/524475/141149856-8d2d63cb-867b-4fa2-9b3d-9f46ced81e67.png)

## Changes
- Introduce `SettingsViewModel` and `SettingsFactory` and make settings screen follow MVVM architecture.
- Add basic unit tests using `SettingsViewModelTests`
- Fix typos in comments.

## Testing
- Launch the app and navigate to `My Store` screen.
- Navigate to Settings screen by tapping on the top right `gear` button.
- Observe that you can view all settings related to the account.
- Please navigate to child screens such as `Plugins`, `In-Person Payments` and make sure that the navigation works as expected.
- Switching the store should reload the sites in settings screen.

## Notes
- Being an outside contributor, I don't have a payment method set up in my store. I request you to test the child screens including `In-Person Payments` using your store. 

Update release notes:
